<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="model" default="$(find three_leg_robot)/models/robot_6_joint/robot_6_joint.urdf.xacro"/>
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(arg model)" />

  <rosparam file="$(find three_leg_robot)/models/robot_6_joint/motors_config.yaml"/>


   <include file="$(find gazebo_ros)/launch/empty_world.launch"> 
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="debug" value="false"/>
        <!--<arg name="world_name" value="$(arg world_file)"/>-->
  </include>

  <node name="mybot_spawn" pkg="gazebo_ros" type="spawn_model" output="screen" args="-urdf -param robot_description -model my_bot -x 0 -y 0 -z 0"/>

  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="joint_state_controller" ns="/robot_6_joint"/>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" >
    <remap from="joint_states" to="/robot_6_joint/joint_states"/>
    </node>

  
</launch>
